src/syntree.rs

```diff
@@ -407,9 +555,12 @@ fn parse_type<'t>(ctx: Context<'t>) -> ParseResult<'t, Type> {
         }
     };
 
+    // Wrap it in a syntax tree node.
     let ty = Type { span, kind };
 
+    // Union type, a | b
```

Comment:
```suggestion
    // Union type, `a | b`
```
Some small inconsistencies around this. I don't know if I caught all of them.